src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/SymbolType.cs

```diff
@@ -271,18 +271,30 @@ public override Type MakeByRefType()
             return FormCompoundType(_format + "&", _baseType, 0)!;
         }
 
+        [RequiresDynamicCode("The code for an array of the specified type might not be available.")]
         public override Type MakeArrayType()
         {
             return FormCompoundType(_format + "[]", _baseType, 0)!;
         }
 
+        [RequiresDynamicCode("The code for an array of the specified type might not be available.")]
         public override Type MakeArrayType(int rank)
         {
-            string s = GetRankString(rank);
+            string s = GetRank(rank);
```

Comment:
```suggestion
            string s = FormatRank(rank);
```
I think it would be a better